### DISCUSSION TITLE
Module file reduction with webpack(Around half of the kb size is expecting rather than present sizes)

### DISCUSSION THREAD

**Discussion by ghost** [webpack/webpack#15326] [Ideas]
Hi Team,

We need webpack configuration for less file size reduction using react components. Currently, we are using terser plugin in our webpack configuration. But, as per our client requirement/project demand, we require reduction in file sizes for webpack better than terser plugin.

For your best reference, please find the attached screenshot to know what is the existed file sizes we are using in our project.

![reductionwebpack](https://user-images.githubusercontent.com/10085269/152736110-b4529bff-a462-46e8-8019-0fc42a8c35cb.png)

If anyone idea on about file size reduction on webpack configuration, please feel free to share your thoughts.

Kind Regards,
Saikrishna.

**Comment by vankop** [comment]
all enabled `optimization` options in `webpack.config` will help to reduce output size.. you can use [webpack-bundle-analyzer](https://www.npmjs.com/package/webpack-bundle-analyzer ) to analyze output

**Reply by ghost** [reply]
Hi [Ivan Kopeykin](https://github.com/vankop),

Can you please help us, we are using below are the optimization options in the current project requirement:

optimization: {
    minimize: true,
    minimizer: [
      // This is only used in production mode
      new TerserPlugin({      
        terserOptions: {
          parse: {
            ecma: 8
          },
          compress: {
            ecma: 5,
            warnings: false,
            comparisons: false,
            inline: 2
          },
          mangle: {
            safari10: true
          },
          output: {
            ecma: 5,
            comments: /^\**!|@qarts_version|Last updated/i,
            // Turned on because emoji and regex is not minified properly using default
            // https://github.com/facebook/create-react-app/issues/2488
            ascii_only: true
          }
        },
        // Enable file caching
        cache: true,
        sourceMap: false // Disable .map files
      }),

 // This is only used in production mode
      new OptimizeCSSAssetsPlugin({
        cssProcessorOptions: {
          parser: safePostCssParser,
          map: {
            // `inline: false` forces the sourcemap to be output into a
            // separate file
            inline: false,
            // `annotation: true` appends the sourceMappingURL to the end of
            // the css file, helping the browser find the sourcemap
            annotation: true
          }
        }
      }),
],
    removeAvailableModules: false,
    removeEmptyChunks: false,
    splitChunks: {
      chunks: "all",
      // name: false,
      name: true,
      cacheGroups: {
        commons: {
          test: /[\\/]node_modules[\\/]/,
          // test: /[\\/]node_modules[\\/](!moment)[\\/]/,
          // name: "vendor-bundle"
          name: `./${QARTS_NAME}_bundle`
        },
      },
    },
    // runtimeChunk: true,
  },
  externals: {
    'react': 'React',
    'react-dom': 'ReactDOM',
  },
  module: {
    rules: [
      {
        oneOf: styleLoaders()
      }
    ]
  },

Please let us know, what are the options will use rather than using above options for optimization. Your guidance will helpful for us.

Kind Regards,
Saikrishna.
